simulation/container.py

```diff
@@ -147,29 +157,50 @@ def simulation(self) -> None:
               and add new dead instance
         """
 
+        print(len(self.object_list))
         while self.is_alive():
+            start = time.time()
+            # Note: Move and can_infect parameter reset. Then sort.
             for instance in self.object_list:
                 x = random.uniform(-self.move_distance_length, self.move_distance_length)
                 y = random.uniform(-self.move_distance_length, self.move_distance_length)
                 instance.move(x, y, self.width, self.height)
+                instance.can_infect = True
+
+            self.object_list.sort(key=lambda parameter: parameter.x)
+            # --- end.
 
+            for counter, instance in enumerate(self.object_list, 1):
                 if instance.current_condition == "susceptible":
-                    for infected in self.object_list:
-                        if infected.current_condition == "infected":
-                            if instance.is_in_infection_area(infected.x, infected.y,
-                                                             infected.infection_range):
+                    for next_instance in itertools.islice(self.object_list, counter, len(self.object_list)):
+                        if next_instance.current_condition == "infected":
+                            if instance.is_in_infection_area(next_instance.x, next_instance.y,
+                                                             next_instance.infection_range):
                                 if instance.infect():
+                                    instance.can_infect = False
```

Comment:
There is no need to set this parameter for current susceptible individual even when it will be marked as infected now. We will not be checking it again in further iterations.